independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanProcessor.java

```diff
@@ -223,16 +232,17 @@ public BeanDeployment getBeanDeployment() {
     }
 
     public BeanDeployment process() throws IOException {
-        registerCustomContexts();
-        registerBeans();
-        registerSyntheticObservers();
-        initialize(new Consumer<BytecodeTransformer>() {
+        Consumer<BytecodeTransformer> unsupportedBytecodeTransformer = new Consumer<BytecodeTransformer>() {
             @Override
             public void accept(BytecodeTransformer transformer) {
-
+                throw new UnsupportedOperationException();
```

Comment:
Why unsupported as opposed to just no-op? Also a comment would be nice to know that this is only for test container and is not going to be actually invoked.

Reply:
"no-op" would just cause a problem later... In theory, it can be used for non-test scenarios as well. From my POV it's just illegal state if `transformUnproxyableClasses=true` and no valid `Consumer<BytecodeTransformer>` is provided.

Reply:
>  it's just illegal state if transformUnproxyableClasses=true and no valid Consumer<BytecodeTransformer> is provided.

Fair enough, that makes sense.